reactos/drivers/usb/usbport/usbport.c

```diff
@@ -2275,21 +2283,11 @@ USBPORT_MapTransfer(IN PDEVICE_OBJECT FdoDevice,
     Transfer->SgList.CurrentVa = CurrentVa;
 
     Mdl->MdlFlags |= MDL_MAPPING_CAN_FAIL;
-
-    if (Mdl->MdlFlags & (MDL_SOURCE_IS_NONPAGED_POOL |
-                         MDL_MAPPED_TO_SYSTEM_VA))
-    {
-        MappedSystemVa = Mdl->MappedSystemVa;
-    }
-    else
-    {
-        MappedSystemVa = MmMapLockedPages(Mdl, KernelMode);
-    }
+    MappedSystemVa = MmGetSystemAddressForMdl(Mdl);
```

Comment:
The -Safe version also removes the need to change MdlFlags (which is also not allowed in drivers)